File: Nivel_5/Retos/DividirDosNumeros.py
num1 = int(input("Ingrese un número entero: "))
num2 = int(input("Ingrese otro número entero: "))
try:
    resultado = num1 / num2  # División
    print(f"El resultado de la división es: {resultado}")
except ZeroDivisionError: print("Error: División por cero no permitida.")
except ValueError: print("Error: Debes introducir un número entero.")
finally:
    print("Programa finalizado.")

File: Nivel_5/Retos/DividirDosNumeros.java
package Nivel_5.Retos;

import java.util.*;

public class DividirDosNumeros {
    public static void main(String[] args) {
        Scanner kbd = new Scanner(System.in);
        try {
            System.out.print("Ingrese un número entero: ");
            int num1 = kbd.nextInt();
            System.out.print("Ingrese otro número entero: ");
            int num2 = kbd.nextInt();
            int resultado = num1 / num2; // División
            System.out.println("El resultado de la división es: " + resultado);
        } catch (InputMismatchException  e) {System.out.println("Error: Debes introducir un número entero.");
        } catch (ArithmeticException e) {System.out.println("Error: División por cero no permitida.");}
        finally {
            kbd.close();
            System.out.println("Programa finalizado.");
        }
    }
}
